src/solc_0.7/catalyst/AssetMinter.sol

```diff
@@ -116,6 +120,7 @@ contract AssetMinter is WithMetaTransaction {
                 uint16 maxGems = _gemsCatalystsRegistry.getMaxGems(assets[i].catalystId);
                 require(assets[i].gemIds.length <= maxGems, "INVALID_GEMS_TOO_MANY");
                 catalystsQuantities[assets[i].catalystId]--;
+                // @review why assign a new decremented value to gemsQuantities here if it's not used again?
```

Comment:
It is checked above in the loop, so if it reach zero it stop.
it could actually underflow so we should actually ensure it does not, either via check manually or using SafeMath

Reply:
noted.